src/javascript/SelectorTypes/Picker/Picker.configs.js

```diff
@@ -74,111 +74,147 @@ const treeConfigs = {
-        listTypesTable: ['jnt:file'],
-        selectableTypesTable: ['jnt:file']
-    },
-    user: {
-        picker: ContentPickerSelectorType,
-        treeConfigs: [treeConfigs.users, treeConfigs.siteUsers],
-        searchSelectorType: 'jnt:user',
-        searchPaths: site => ['/users', `/sites/${site}/users`],
-        listTypesTable: ['jnt:user'],
-        selectableTypesTable: ['jnt:user'],
-        displayTree: false
-    },
-    usergroup: {
-        picker: ContentPickerSelectorType,
-        treeConfigs: [treeConfigs.groups, treeConfigs.siteGroups],
-        searchSelectorType: 'jnt:group',
-        searchPaths: site => ['/groups', `/sites/${site}/groups`],
-        listTypesTable: ['jnt:group'],
-        selectableTypesTable: ['jnt:group'],
-        displayTree: false
-    },
-    category: {
-        picker: ContentPickerSelectorType,
-        treeConfigs: [treeConfigs.categories],
-        searchSelectorType: 'jnt:category',
-        listTypesTable: ['jnt:category'],
-        selectableTypesTable: ['jnt:category']
-    },
-    site: {
-        picker: ContentPickerSelectorType,
-        treeConfigs: [treeConfigs.sites],
-        searchSelectorType: 'jnt:virtualsite',
-        listTypesTable: ['jnt:virtualsite'],
-        selectableTypesTable: ['jnt:virtualsite']
-    },
-    getPickerSelectorType(options) {
-        return this[this._getPickerType(options)].picker;
-    },
-    resolveConfig(options, field) {
-        const config = Object.assign({}, this[this._getPickerType(options)]);
-        if (field && field.valueConstraints) {
-            const constraints = field.valueConstraints.map(constraint => constraint.value.string);
-            if (constraints && constraints.length > 0) {
-                config.selectableTypesTable = constraints;
-            }
-        }
 
-        return config;
-    },
-    _getPickerType(options) {
-        const option = options && options.find(option => option.name === 'type' && this[option.value]);
-        return (option && option.value) || 'editorial';
+const _getPickerType = options => {
+    const pickerOption = options && options.find(option => option.name === 'type' && registry.get('pickerConfiguration', option.value));
+    return (pickerOption && pickerOption.value) || 'editorial';
+};
+
+export const resolveConfig = (options, field) => {
+    const config = Object.assign({}, registry.get('pickerConfiguration', _getPickerType(options)).cmp);
+    if (field && field.valueConstraints) {
+        const constraints = field.valueConstraints.map(constraint => constraint.value.string);
+        if (constraints && constraints.length > 0) {
+            config.selectableTypesTable = constraints;
+        }
     }
+
+    return config;
 };
 
-export default pickerConfigs;
+export const getPickerSelectorType = options => {
+    return registry.get('pickerConfiguration', _getPickerType(options)).cmp.picker;
+};
+
+export const registerPickerConfig = ceRegistry => {
```

Comment:
I see that we put all selectors at the root of the registry with global names, it would have be better to namespace them for Content editor.